azerothcore/azerothcore-wotlk
# fix(Core/Condition): Implement CONDITION_QUESTSTATE

For you my love @BarbzYHOOL 





## CHANGES PROPOSED:
* Implement CONDITION_QUESTSTATE

* Imported from TrinityCore 


## ISSUES ADDRESSED:

Ref: https://github.com/azerothcore/azerothcore-wotlk/issues/1372

## TESTS PERFORMED:
- Builds



## HOW TO TEST THE CHANGES:

```sql
DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=19 AND `SourceGroup`=0 AND `SourceEntry`=40;
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES 
(19, 0, 40, 0, 0, 47, 0, 47, 8, 0, 0, 0, 0, '', '');
```
THIS IS A DUMMY SQL. MAKE SURE TO DELETE IT AFTER TESTING.

1. Apply SQL
2. .go c id 241
3. According to what your ConditionValue2 is, see it work ingame.
4. The "8" in the script above can be changed to any of these values. IT IS A BITMASK SO YOU CAN COMBINE MULTIPLE
(1 = not taken, 2 = completed, 8 = in progress, 32 = failed, 64 = rewarded)
5. You only really need to test one
## KNOWN ISSUES AND TODO LIST:



## Target branch(es):








 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR
